test/support/reduce_serial_impl.h

```diff
@@ -0,0 +1,50 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _REDUCE_SERIAL_IMPL_H
+#define _REDUCE_SERIAL_IMPL_H
+
+#include <iterator>
+
+template <typename ViewKeys, typename ViewVals, typename ResKeys, typename ResVals, typename Size,
+          typename BinaryPredicate, typename BinaryOperation>
+Size
+reduce_by_segment_serial(ViewKeys keys, ViewVals vals, ResKeys& res_keys, ResVals& res_vals, Size n,
```

Comment:
`ViewKeys keys, ViewVals vals` are passed by value...  What type of passed parameter? An iterator or std::vector?
I see that res_keys and res_vals - `std::vector`...
I would suggest to keep consistent here  - f.e. to pass the iterators for input and output sequences.

Reply:
`keys` and `vals` are iterators. I have made the change to pass `res_keys` and `res_vals` by value as iterators and renamed the iterator types accordingly.